src/bp-messages/classes/class-bp-messages-message.php

```diff
@@ -153,7 +153,14 @@ public function send() {
 
 		// If we have no thread_id then this is the first message of a new thread.
 		if ( empty( $this->thread_id ) ) {
-			$this->thread_id = (int) $wpdb->get_var( "SELECT MAX(thread_id) FROM {$bp->messages->table_name_messages}" ) + 1;
+			$max_thread      = BP_Messages_Message::get(
+				array(
+					'fields'   => 'thread_ids',
+					'per_page' => 1,
+					'page'     => 1,
+				)
+			);
+			$this->thread_id = (int) $max_thread['messages'][0] + 1;
```

Comment:
@janakkaneriya007 We need to check Query returning result or not.
```suggestion
                         $this->thread_id = 1;
			if ( ! empty( $max_thread['messages'][0] ) ){
			       $this->thread_id = (int) $max_thread['messages'][0] + 1;
			}
```